src/xtgeo/io/_file_wrapper.py

```diff
@@ -0,0 +1,618 @@
+"""A FileWrapper class to wrap around files and streams representing files."""
+from __future__ import annotations
+
+import io
+import os
+import pathlib
+import platform
+import re
+import struct
+import uuid
+from os.path import join
+from tempfile import mkstemp
+from typing import TYPE_CHECKING, Literal, Union
+
+import h5py
+
+import xtgeo._cxtgeo
+from xtgeo.common import null_logger
+
+logger = null_logger(__name__)
+
+if TYPE_CHECKING:
+    import pathlib.Path
+
+    from xtgeo.cube import Cube
+    from xtgeo.grid3d import Grid, GridProperties, GridProperty
+    from xtgeo.surface import RegularSurface, Surfaces
+    from xtgeo.wells import BlockedWell, BlockedWells, Well, Wells
+    from xtgeo.xyz import Points, Polygons
+
+    XTGeoObject = Union[
+        BlockedWell,
+        BlockedWells,
+        Cube,
+        Grid,
+        GridProperty,
+        GridProperties,
+        Points,
+        Polygons,
+        RegularSurface,
+        Surfaces,
+        Well,
+        Wells,
+    ]
+
+SUPPORTED_FORMATS = {
+    "rmswell": ["rmswell", "rmsw", "w", "bw"],
+    "roff_binary": ["roff_binary", "roff", "roff_bin", "roff-bin", "roffbin", "roff.*"],
+    "roff_ascii": ["roff_ascii", "roff_asc", "roff-asc", "roffasc", "asc"],
+    "egrid": ["egrid"],
+    "fegrid": ["fegrid"],
+    "init": ["init"],
+    "finit": ["finit"],
+    "unrst": ["unrst"],
+    "funrst": ["funrst"],
+    "grdecl": ["grdecl"],
+    "bgrdecl": ["bgrdecl"],
+    "irap_binary": ["irap_binary", "irap_bin", "rms_binary", "irapbin", "gri"],
+    "irap_ascii": ["irap_ascii", "irap_asc", "rms_ascii", "irapasc", "fgr"],
+    "hdf": ["hdf", "hdf5", "h5"],
+    "segy": ["segy", "sgy", "segy.*"],
+    "storm": ["storm"],
+    "zmap_ascii": ["zmap", "zmap+", "zmap_ascii", "zmap-ascii", "zmap-asc", "zmap.*"],
+    "ijxyz": ["ijxyz"],
+    "petromod": ["pmd", "petromod"],
+    "xtg": ["xtg", "xtgeo", "xtgf", "xtg.*"],
+    "xyz": ["xyz", "poi", "pol"],
+    "rms_attr": ["rms_attr", "rms_attrs", "rmsattr.*"],
+}
+
+VALID_FILE_ALIASES = ["$fmu-v1", "$md5sum", "$random"]
+
+
+class FileWrapper:
```

Comment:
Moved from `src/xtgeo/common/sys.py` ? No changes?

Reply:
Yep, no changes, just a few added mypy ignores since this file is checked now

```python
            short = obj.metadata.opt.shortname.lower().replace(" ", "_")  # type: ignore
            desc = obj.metadata.opt.description.lower().replace(" ", "_")  # type: ignore
            date = obj.metadata.opt.datetime  # type: ignore
```